sdk/typescript/src/providers/json-rpc-provider.ts

```diff
@@ -204,6 +205,23 @@ export class JsonRpcProvider extends Provider {
     return objects.filter((obj: SuiObjectInfo) => Coin.isSUI(obj));
   }
 
+  getCoinDenominationInfo(
+    coin_type: string,
+  ): CoinDenominationInfoResponse {
+    if (coin_type !== '0x2::sui::SUI') {
+      throw new Error(
+        'only SUI coin is supported in getCoinDenominationInfo for now.'
+      );
+    }
```

Comment:
This will break if user passes `0x00....002::sui::SUI` which also has a valid package id
```suggestion
    coinType: string
  ): CoinDenominationInfoResponse {
    const [package, module, symbol] = coinType.split('::');
    if (normalizeSuiAddress(packageId) !== normalizeSuiAddress('0x2') || 'module' != 'sui' || 'symbol' !== 'SUI') {
      throw new Error(
        'only SUI coin is supported in getCoinDenominationInfo for now.'
      );
    }
```